### PR TITLE
PUB-1569 Updated cron job and logic

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1569

### Change description ###

- Updated cron job for media application reporting to run every Monday at 12am
- Updated logic so the media application reporting email request is only sent if its not empty

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by ChrisS1512 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java [hunk 1]
```diff
@@ -147,19 +147,15 @@ public void deleteApplication(UUID id) {
...
-
-    /**
-     * Collate media applications and send them for reporting.
-     */
     public void processApplicationsForReporting() {
         List<MediaApplication> mediaApplications = getApplications();
-        log.info(publicationService.sendMediaApplicationReportingEmail(mediaApplications));
+        if (!mediaApplications.isEmpty()) {
+            log.info(publicationService.sendMediaApplicationReportingEmail(mediaApplications));
+            processApplicationsForDeleting();
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java:157]: We've got separate endpoints for deleting / reporting on media applications - those will need to be combined if we are doing deleting / reporting in one method, or the deleting needs to be split out from the above

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java (hunk 1 above)
Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java:157]: Will combine as it should have always been in one method, it always reports then deletes

Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java:157]: Updated the controller so there is only one which executes both tasks (will only happen if there are applications to process).

**Review by ChrisS1512 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java (hunk 1 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java:157]: Looks good! I've approved this now

**Review by KianKwa — APPROVED** [review]

**Pull request merged**